lib/mongo/grid/stream/read.rb

```diff
@@ -167,7 +167,8 @@ def ensure_readable!
           end
 
           def view
-            @view ||= fs.chunks_collection.find({ :files_id => file_id }, options).read(read_preference).sort(:n => 1)
+            @view ||= (opts = options.merge(read: read_preference) if read_preference
```

Comment:
The only thing that comes to mind is putting those two lines in a separate function and then conditionally-assigning the result of calling it to view. That may or may not be worth it depending on whether you want keep all of the logic together rather than separating it out, though.